src/tests/ftest/launch.py

```diff
@@ -901,8 +905,18 @@ def archive_logs(avocado_logs_dir, test_yaml, args):
 
     # Copy any log files written to the DAOS_TEST_LOG_DIR directory
     logs_dir = os.environ.get("DAOS_TEST_LOG_DIR", DEFAULT_DAOS_TEST_LOG_DIR)
-    archive_files(destination, host_list, "{}/*.log".format(logs_dir))
 
+    # Caution: the glob expression "_output.log" must match the
+    #   --output-filename specified in # cart_utils.py:get_env()
+
+    # Create tar archives of orterun log dirs to prevent name collisions (e.g.,
+    # we'll have many # hundreds of tests with output files named "stdout")
+    archive_files(destination, host_list,
+      "{}/*_output.orterun_log".format(logs_dir), do_tar=True)
+
+    # Plain files need not be tar'd, then can simply be scp'd to the archive
+    # destination
+    archive_files(destination, host_list, "{}/*.{{tar*,log*}}".format(logs_dir))
```

Comment:
As part of the CI process there is already code in place that will [compress any file larger than 1M](https://github.com/daos-stack/daos/blob/master/ci/functional/job_cleanup.sh#L24) after the execution of `launch.py`/`ftest.sh`.  There is also ongoing work in https://github.com/daos-stack/daos/pull/2029 to move this compression to a locally run script that will compress the files prior to remote-copying them back to the host running `launch.py` (essentially replacing the work done inside `archive_files`).

With all of that in consideration, I would suggest simplifying the cart remote log handling to fit in this model by having cart_utils.py produce `*.log` files (if it isn't already - lines 910 and 915 seem to contradict each other), revert back to the `archive_files(destination, host_list, "{}/*.log".format(logs_dir))` call, and allow the Jenkins scripts (i.e. `ci/functional/job_cleanup.sh`) to handle any log file compression.  When PR 2029 completes this will be enhanced for all logs and no special handling will be required.

In addition, using the 1M compression threshold is useful because small logs can be viewed within the browser and do not need to be downloaded and uncompressed to be viewed.

Reply:
Hi @phender,

Thank you for your comments and review!

A few questions/comments below.

> As part of the CI process there is already code in place that will compress
> any file larger than 1M after the execution of launch.py/ftest.sh. There is
> also ongoing work in #2029 to move this compression to a locally run script
> that will compress the files prior to remote-copying them back to the host
> running launch.py (essentially replacing the work done inside archive_files).

The tar option to `archive_files` isn't for compression, but to avoid filename collisions in orterun logs. E.g., the following directory structure is duplicated for each orterun command, so tar-ing them up avoids a complex renaming scheme.

```
  .../@test-name@.orterun_log/rank.0
  .../@test-name@.orterun_log/rank.0/stdout
  .../@test-name@.orterun_log/rank.0/stderr
  .../@test-name@.orterun_log/rank.1
  .../@test-name@.orterun_log/rank.1/stdout
  .../@test-name@.orterun_log/rank.1/stderr
...
```

Or I'm misunderstanding how we can fold the above functionality into [#2029](https://github.com/daos-stack/daos/pull/2029).

> 
> With all of that in consideration, I would suggest simplifying the cart
> remote log handling to fit in this model by having cart_utils.py produce
> *.log files (if it isn't already - lines 910 and 915 seem to contradict each
> other), revert back to the archive_files(destination, host_list,
> "{}/*.log".format(logs_dir)) call, and allow the Jenkins scripts (i.e.
> ci/functional/job_cleanup.sh) to handle any log file compression. When PR
> 2029 completes this will be enhanced for all logs and no special handling
> will be required.

Looks like `job_cleanup.sh` is run after all the functional tests, but because `launch.py` is run with the `-c` ("clean logs") option -- won't the logs we want to archive get removed before `job_cleanup.sh` runs?

> In addition, using the 1M compression threshold is useful because small logs
> can be viewed within the browser and do not need to be downloaded and
> uncompressed to be viewed.

👍

Reply:
@mallove79 thank you for the clarification.

So `launch.py`, with the `--archive, -a` argument, is used to get files generated on **remote host** by the test back to the launch node to be included in the avocado job_results directory - otherwise these files are not saved in the CI artifacts.  

Since these remote files are generally similarly named, e.g. `daos.log`, on each host the `launch.py` archive process amends the original file name with the name of the server from which it was copied - to retain file name uniqueness.  These files are placed in the `daos_logs` sub directory for each test.   If a test runs multiple variants, like `ftest/daos_test/daos_core_test.py` - through the [mux'ing of test yaml paramters](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/daos_test/daos_core_test.yaml#L56) - then the remote files generated by the execution of each test variant will need to be unique as the `launch.py` archive step is only run after all the single test's variants are complete (the execution of a single `avocado run ...` command).  The `ftest/daos_test/daos_core_test.py` handles this by generating [test-variant-specific](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/2858/artifact/Functional/job-2020-09-18T13.49-aa7c4aa-daos_test-daos_core_test/daos_logs/) log file names with this [code](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/daos_core_base.py#L114).

Now if we are taking about locally generated log file names, the avocado test framework provides the [`self.outputdir`](https://avocado-framework.readthedocs.io/en/52.0/api/core/avocado.core.html?highlight=outputdir#avocado.core.test.Test.outputdir) directory.  The `self.outputdir` places files in a test-variant specific job-results / CI artifact test sub directory: `job-<YYYY>-<MM>-<DD>T<HH>.<MM>-<test_id>/test_results/<job_num>-.<test_variant>/data` (e.g. this [cmoka results xml file](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/2858/artifact/Functional/job-2020-09-18T13.49-aa7c4aa-daos_test-daos_core_test/test-results/01-._daos_test_daos_core_test.py%3ADaosCoreTest.test_subtest%3Btransport_config-test_d-num_clients-num_replicas-transport_config-hosts-pool-0-1-transport_config-337e/data/) generated by the same test)

The `launch.py` clean step (`--clean, -c`) is run prior to each test to ensure any remote log file that failed to archive from a previous run does not get archived with the next test's files.  This assumes all of these log files match `*.log`.

Where are the `@test-name@.orterun_log/rank.*` files being generated on the launch node or on remote nodes?

Reply:
We'll remove the offending changes here, and plan to fix the log arching in [this PR](https://jira.hpdd.intel.com/browse/DAOS-5662).